python/triton/code_gen.py

```diff
@@ -726,7 +741,10 @@ def __call__(self, *wargs, **kwargs):
 
 
 class Autotuner:
-    def __init__(self, kernel, arg_names, configs, key, reset_to_zero):
+    def __init__(self, kernel, arg_names, configs, key, reset_to_zero, prune_configs_by=None):
```

Comment:
I would add some typehints here, and also it's probably time for a docstring :) otherwise it will be harder for users to pass the right format for `prune_configs_by`. Maybe we want a datastructure that's more explicit than a dict for this?